### PR TITLE
now can handle null d attribute

### PR DESCRIPTION
**Pull request by jfkonecn**
Fix #9251
[here's the link](https://github.com/cBioPortal/cbioportal/issues/9251)

@dippindots 

I changed the case on svgpath since this repo wasn't building on ubuntu for me. It looks like the original project solved the [same problem](https://github.com/yWorks/svg2pdf.js/issues/91) the same way.

I added a null check for the node.getAttribute("d") since it can return null, and when it does return null it causes an exception. Similarly the original project has the same [guard](https://github.com/yWorks/svg2pdf.js/blob/master/src/nodes/path.ts#L23) against nulls.

### CODE REVIEW

File: src/svg2pdf.js [hunk 1]
```diff
@@ -2260,7 +2260,7 @@ SOFTWARE.
     });
   } else if (typeof module !== "undefined" && module.exports) {
     RGBColor = require("./rgbcolor.js");
-    SvgPath = require("SvgPath");
+    SvgPath = require("svgpath");
```
Comment (dippindots) [inline src/svg2pdf.js:2263]: @jfkonecn Thanks for fixing this issue! Do we need to change the `SvgPath` to `svgpath` too on line `2254`? I see they changed that too at here: https://github.com/yWorks/svg2pdf.js/pull/92/files#diff-93163a5a97c109e72dcc8a56277716b5792790def1e1f21e9f8ec51144d5cd8cR2319

**Comment by dippindots** [comment]
I tested it locally and it works, I can now download the pdf without any issues. But I noticed that the x-axis of the chart is weird, I will make another ticket for it.
image

File: src/svg2pdf.js (hunk 1 above)
Comment (jfkonecn) [reply src/svg2pdf.js:2263]: I'll fix that too

**Comment by jfkonecn** [comment]
> I tested it locally and it works, I can now download the pdf without any issues. But I noticed that the x-axis of the chart is weird, I will make another ticket for it.
> 
> image
> 
> 

Should I try pulling in their latest code? I know there are a lot of changes, but it might be worth it to fix all the bugs.

File: src/svg2pdf.js (hunk 1 above)
Comment (dippindots) [reply src/svg2pdf.js:2263]: Thank you!

**Comment by dippindots** [comment]
@jfkonecn Please feel free to try to update more, although I don't think it would be an easy task, if it's too complex then we can think about how to fix this (maybe fix it in the frontend codebase).

**Comment by jfkonecn** [comment]
Sounds good! I'll give it a shot and let you know if it's too difficult.

**Comment by jfkonecn** [comment]
Made a [fix](https://github.com/cBioPortal/svg2pdf.js/pull/2), but it does seem that you no longer need to fork this repository. I didn't have to change any of the core logic to get the cbioportal tests to pass.

**Comment by dippindots** [comment]
Hi, @jfkonecn thanks for trying to pull more commits from the original project in your new pull request: https://github.com/cBioPortal/svg2pdf.js/pull/2. I remember the reason we decided not to use the original svg2pdf is: if we use their latest version, they require us to update our jspdf and if we also update jspdf, then it's not compatible with our project.
Have you tried this in the cbioportal frontend project?

**Comment by jfkonecn** [comment]
> Hi, @jfkonecn thanks for trying to pull more commits from the original project in your new pull request: #2. I remember the reason we decided not to use the original svg2pdf is: if we use their latest version, they require us to update our jspdf and if we also update jspdf, then it's not compatible with our project. Have you tried this in the cbioportal frontend project?

I just tried it and the pdf export is generating the same error as what was reported in the bug report. 
Here's the a [draft PR](https://github.com/cBioPortal/cbioportal-frontend/pull/4597) for it.
Let me know what you want to do about it.

**Comment by dippindots** [comment]
@jfkonecn Thanks for testing this, I think maybe we can merge this pull request into our forked repo and update the frontend since this issue was also not fixed in their original project.

**Comment by jfkonecn** [comment]
sounds good to me

**Comment by dippindots** [comment]
@jfkonecn Thanks again for helping fix this issue! Merged. I will create a new tag and update the frontend.

**Pull request merged**